hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/FileGroupReaderBasedMergeHandle.java

```diff
@@ -220,6 +323,52 @@ public void doMerge() {
     }
   }
 
+  private void setRecordLocations(HoodieRecord<T> record) {
+    if (recordIndexCallbackOpt.isEmpty()) {
```

Comment:
if the location is only used for RLI metadata inference, we can return early when `HoodieRecord.getIgnoreIndexUpdate` is true.